src/cmd/flux-job.c

```diff
@@ -1893,6 +1896,28 @@ static void status_eventlog_cb (flux_future_t *f, void *arg)
         else
             stat->exit_code = WEXITSTATUS (stat->status);
     }
+    else if (!strcmp (name, "exception")) {
+        const char *type;
+        int severity;
+        const char *note = NULL;
+
+        if (json_unpack (context, "{s:s s:i s?:s}",
+                         "type", &type,
+                         "severity", &severity,
+                         "note", &note) < 0)
+            log_err_exit ("error decoding exception context");
+        if (severity == 0) {
+            /* Note: the exit_code and status will be overridden
+             *  by the finish event if this job is still running
+             *  o/w, for a non-running job with a fatal exception
+             *  the default exit code is simply 1.
+             */
+            stat->status = 256;
+            stat->exit_code = 1;
+            stat->exception = true;
+            strncpy (stat->ex_type, type, sizeof(stat->ex_type) - 1);
```

Comment:
(Not a request for a change, but a question for my own understanding and edification)

If the length of `type` is > `sizeof(stat->ex_type) - 1` and `stat` wasn't zero'd when alloc'd, then `stat->ex_type` will not be NULL-terminated, right?  Tracing back to [`stat`'s allocation, it looks like it was `calloc`d](https://github.com/grondo/flux-core/blob/04abb6288b3812154de4a2e29223e9f624be727a/src/cmd/flux-job.c#L1947), so I don't think it is ultimately an issue here.   For my own understanding though, is best-practice to NULL-terminate the string "just to be safe"?  Or is it to fine to rely on structs and `char[]`s to have been `calloc`'d so that you don't have to worry about this edge-case?

Reply:
I think it's often fine to rely on zero-initialized buffers when it's obvious that's the case.